scipy/sparse/linalg/tests/test_propack.py

```diff
@@ -0,0 +1,148 @@
+@pytest.mark.parametrize('irl', (True, False))
+@pytest.mark.parametrize('which', ('LM', 'SM'))
+def test_svdp(ctor, dtype, irl, which):
+    np.random.seed(0)
+    n, m, k = 10, 20, 3
+    if which == 'SM' and not irl:
+        with assert_raises(ValueError):
+            check_svdp(n, m, ctor, dtype, k, irl, which)
+    else:
+        check_svdp(n, m, ctor, dtype, k, irl, which)
+
+
+def load_coord(folder, precision, file="illc1850.coord"):
+    dtype = {"single": np.float32, "double": np.float64}[precision]
+    path = os.path.join(folder, precision, file)
+    with open(path) as f:
+        m, n, nnz = (int(val) for val in f.readline().split())
+        coord = np.array([[float(val) for val in line.split()] for line in f])
+    i = coord[:, 0].astype(int) - 1
+    j = coord[:, 1].astype(int) - 1
+    data = coord[:, 2].astype(dtype)
+    A = coo_matrix((data, (i, j)))
+    return A
+
+
+def load_sigma(folder, precision="double", irl=False):
+    dtype = {"single": np.float32, "double": np.float64}[precision]
+    s_name = "Sigma_IRL.ascii" if irl else "Sigma.ascii"
+    path = os.path.join(folder, precision, s_name)
+    with open(path) as f:
+        data = np.array([float(line.split()[1]) for line in f], dtype=dtype)
+    return data
+
+
+def load_uv(folder, precision="double", uv="U", irl=False):
+    dtype = {"single": np.float32, "double": np.float64}[precision]
+    uv_name = (uv + "_IRL.ascii") if irl else (uv + ".ascii")
+    path = os.path.join(folder, precision, uv_name)
+    with open(path) as f:
+        m, n = (int(val) for val in f.readline().split())
+        data = np.array([float(val.strip()) for val in f], dtype=dtype)
+    return data.reshape((n, m)).T
+
+
+@pytest.mark.parametrize('precision', ('single', 'double'))
+@pytest.mark.parametrize('irl', (False, True))
+def test_examples(precision, irl):
+    atol = {'single': 1e-3, 'double': 1e-11}[precision]
+
+    path_prefix = os.path.dirname(__file__)
+    relative_path = "propack_examples"
+    folder = os.path.join(path_prefix, relative_path)
+
+    A = load_coord(folder, precision)
+    s_expected = load_sigma(folder, precision, irl=irl)
+    u_expected = load_uv(folder, precision, "U", irl=irl)
+    vt_expected = load_uv(folder, precision, "V", irl=irl).T
+
+    k = len(s_expected)
+    u, s, vt = svdp(A, k, irl_mode=irl)
+
+    assert_allclose(s, s_expected, atol)
+    assert_allclose(np.abs(u), np.abs(u_expected), atol=atol)
+    assert_allclose(np.abs(vt), np.abs(vt_expected), atol=atol)
+
+
+@pytest.mark.parametrize('shifts', (None, -10, 0, 1, 10, 70))
+@pytest.mark.parametrize('dtype', (np.float32, np.float64,
+                                   np.complex64, np.complex128))
+def test_shifts(shifts, dtype):
+    np.random.seed(0)
+    n, k = 70, 10
+    A = np.random.random((n, n))
+    if shifts is not None and ((shifts < 0) or (k > min(n-1-shifts, n))):
+        with pytest.raises(ValueError):
+            svdp(A, k, shifts=shifts, irl_mode=True)
```

Comment:
This part is basically just input validation, right?  If so, I'll write a separate test for this (and the rest of the input validation).

Reply:
Yes, these are input validation tests (checking to see that the right error is raised at the right time)